jupyter_server/services/sessions/handlers.py

```diff
@@ -152,6 +154,8 @@ async def patch(self, session_id):
             changes["name"] = model["name"]
         if "type" in model:
             changes["type"] = model["type"]
+        if "env" in model:
```

Comment:
Same question here, shouldn't this be conditional based on the value of `allow_setup_custom_env_variables`?